shared/contracts/BaseTemplate.sol

```diff
@@ -0,0 +1,191 @@
+        Agent agent = Agent(_installNonDefaultApp(dao, AGENT_APP_ID));
+        agent.initialize();
+        return agent;
+    }
+
+    function _createAgentPermissions(ACL acl, Agent agent, address grantee, address manager) internal {
+        acl.createPermission(grantee, agent, agent.EXECUTE_ROLE(), manager);
+        acl.createPermission(grantee, agent, agent.RUN_SCRIPT_ROLE(), manager);
+    }
+
+    /* FINANCE */
+
+    function _installFinanceApp(Kernel dao, Vault vault, uint64 periodDuration) internal returns (Finance) {
+        Finance finance = Finance(_installNonDefaultApp(dao, FINANCE_APP_ID));
+        finance.initialize(vault, periodDuration);
+        return finance;
+    }
+
+    function _createFinancePermissions(ACL acl, Finance finance, address grantee, address manager) internal {
+        acl.createPermission(grantee, finance, finance.CREATE_PAYMENTS_ROLE(), manager);
+        acl.createPermission(grantee, finance, finance.EXECUTE_PAYMENTS_ROLE(), manager);
+        acl.createPermission(grantee, finance, finance.MANAGE_PAYMENTS_ROLE(), manager);
+    }
+
+    /* TOKEN MANAGER */
+
+    function _installTokenManagerApp(Kernel dao, MiniMeToken token, bool transferable, uint256 maxAccountTokens) internal returns (TokenManager) {
+        TokenManager tokenManager = TokenManager(_installNonDefaultApp(dao, TOKEN_MANAGER_APP_ID));
+        token.changeController(tokenManager);
+        tokenManager.initialize(token, transferable, maxAccountTokens);
+        return tokenManager;
+    }
+
+    function _createTokenManagerPermissions(ACL acl, TokenManager tokenManager, address grantee, address manager) internal {
+        acl.createPermission(grantee, tokenManager, tokenManager.MINT_ROLE(), manager);
+        acl.createPermission(grantee, tokenManager, tokenManager.ASSIGN_ROLE(), manager);
+        acl.createPermission(grantee, tokenManager, tokenManager.REVOKE_VESTINGS_ROLE(), manager);
+    }
+
+    /* VAULT */
+
+    function _installVaultApp(Kernel dao) internal returns (Vault) {
+        Vault vault = Vault(_installDefaultApp(dao, VAULT_APP_ID));
+        vault.initialize();
+        return vault;
+    }
+
+    function _createVaultPermissions(ACL acl, Vault vault, address grantee, address manager) internal {
+        acl.createPermission(grantee, vault, vault.TRANSFER_ROLE(), manager);
+    }
+
+    /* VOTING */
+
+    function _installVotingApp(Kernel dao, MiniMeToken token, uint64 support, uint64 acceptance, uint64 duration) internal returns (Voting) {
+        Voting voting = Voting(_installNonDefaultApp(dao, VOTING_APP_ID));
+        voting.initialize(token, support, acceptance, duration);
+        return voting;
+    }
+
+    function _createVotingPermissions(ACL acl, Voting voting, address grantee, address manager) internal {
+        acl.createPermission(grantee, voting, voting.MODIFY_QUORUM_ROLE(), manager);
+        acl.createPermission(grantee, voting, voting.MODIFY_SUPPORT_ROLE(), manager);
+    }
```

Comment:
We may also want to add helpers to install the Survey app (not sure if we have plans to use it in a production kit)